crates/wiggle/src/lib.rs

```diff
@@ -237,6 +229,7 @@ unsafe impl<T: ?Sized + GuestMemory> GuestMemory for Arc<T> {
 /// already-attached helper methods.
 pub struct GuestPtr<'a, T: ?Sized + Pointee> {
     mem: &'a (dyn GuestMemory + 'a),
+    bc: &'a BorrowChecker,
```

Comment:
FWIW this makes `sizeof(GuestPtr<T>)` to be 4 words large (2 pointers for mem, 1 for bc, one for `pointer`).

It might be best to package up with:

```rust
pub struct BikeshedMe<'a> {
    mem: &'a (dyn GuestMemory + 'a),
    bc: &'a BorrowChecker,
}

pub struct GuestPtr<'a, T> {
    data: &'a BikeshedMe<'a>,
    pointer: T::Pointer,
}
```

(and have `GuestPtr::new` take `&BikeshedMe`)

Reply:
Oh, that's a cool suggestion! :+1:

Reply:
Good point, I prefer your second solution that makes it a method on GuestMemory - the same code is responsible for constructing and passing each.